tests/manage/z_cluster/nodes/test_node_replacement_reactive_aws_ipi.py

```diff
@@ -0,0 +1,206 @@
+            ),
+            pytest.param(
+                *['rbd', 'network failure'],
+                marks=pytest.mark.polarion_id("OCS-2120")
+            ),
+            pytest.param(
+                *['cephfs', 'power off'],
+                marks=pytest.mark.polarion_id("OCS-2119")
+            ),
+            pytest.param(
+                *['cephfs', 'network failure'],
+                marks=pytest.mark.polarion_id("OCS-2121")
+            ),
+        ]
+    )
+    def test_node_replacement_reactive_aws_ipi(
+        self, nodes, pvc_factory, pod_factory, dc_pod_factory,
+        failure, interface
+    ):
+        """
+        Knip-894 Node replacement - AWS-IPI-Reactive
+
+        """
+        # Get worker nodes
+        initial_nodes = get_worker_nodes()
+
+        # Get OSD running nodes
+        osd_running_nodes = get_osd_running_nodes()
+        log.info(f"OSDs are running on nodes {osd_running_nodes}")
+
+        # Label osd nodes with fedora app
+        label_worker_node(osd_running_nodes, label_key='dc', label_value='fedora')
+
+        # Create DC app pods
+        log.info("Creating DC based app pods")
+        if interface == 'rbd':
+            interface = constants.CEPHBLOCKPOOL
+        elif interface == 'cephfs':
+            interface = constants.CEPHFILESYSTEM
+        dc_pod_obj = []
+        for i in range(2):
+            dc_pod = dc_pod_factory(
+                interface=interface, node_selector={'dc': 'fedora'})
+            pod.run_io_in_bg(dc_pod, fedora_dc=True)
+            dc_pod_obj.append(dc_pod)
+
+        # Get app pods running nodes
+        dc_pod_node_name = get_app_pod_running_nodes(dc_pod_obj)
+        log.info(f"DC app pod running nodes are {dc_pod_node_name}")
+
+        # Get both osd and app pod running node
+        common_nodes = get_both_osd_and_app_pod_running_node(
+            osd_running_nodes, dc_pod_node_name
+        )
+        log.info(f"Both OSD and app pod is running on nodes {common_nodes}")
+
+        # Get the machine name using the node name
+        machine_name = machine.get_machine_from_node_name(common_nodes[0])
+        log.info(f"{common_nodes[0]} associated machine is {machine_name}")
+
+        # Get the machineset name using machine name
+        machineset_name = machine.get_machineset_from_machine_name(
+            machine_name
+        )
+        log.info(
+            f"{common_nodes[0]} associated machineset is {machineset_name}"
+        )
+
+        # Get the failure node obj
+        failure_node_obj = get_node_objs(node_names=[common_nodes[0]])
+
+        # Induce failure on the selected failure node
+        log.info(f"Inducing failure on node {failure_node_obj[0].name}")
+        if failure == "power off":
+            # Power off AWS worker node instance
+            nodes.stop_nodes(failure_node_obj, wait=True)
+            log.info(f"Successfully powered off node: {failure_node_obj[0].name}")
+        elif failure == "network failure":
```

Comment:
Can there be more failures? Would another `else` branch make sense?

Reply:
Yes, Network failure is one of the way to make node into "Not Ready". There can be other ways to bring it down.